sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/LogsHelper.cs

```diff
@@ -62,12 +63,28 @@ internal static string GetMessage(LogRecord logRecord)
             {
                 message = logRecord.State.ToString();
             }
-
-            // TODO: Parse logRecord.StateValues to extract originalformat / telemetry properties.
+            else if (logRecord.StateValues != null)
+            {
+                var orginalFormatState = logRecord.StateValues.Where(s => s.Key == "{OriginalFormat}").FirstOrDefault();
+                // If {OriginalFormat} is not found in StateValues
+                // SingleOrDefault will create a new KeyValuePair object with the Key and the Value to NULL
+                message = orginalFormatState.Value?.ToString();
+            }
 
             return message;
         }
 
+        internal static void SetLogStateValuesToProperties(LogRecord logRecord, IDictionary<string, string> properties)
+        {
+            if (logRecord.StateValues != null)
+            {
+                for (int i = 0; i < logRecord.StateValues.Count; i++)
+                {
+                    properties.Add(logRecord.StateValues[i].Key, logRecord.StateValues[i].Value.ToString());
```

Comment:
You may want to exclude {OriginalFormat} else it'll duplicate the info in Message and prop.